drivers/scale_host.go

```diff
@@ -0,0 +1,388 @@
+package drivers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	v1client "github.com/rancher/go-rancher/client"
+	"github.com/rancher/go-rancher/v2"
+	rConfig "github.com/rancher/webhook-service/config"
+	"github.com/rancher/webhook-service/model"
+)
+
+type ScaleHostDriver struct {
+}
+
+func (s *ScaleHostDriver) ValidatePayload(conf interface{}, apiClient client.RancherClient) (int, error) {
+	var key, value string
+
+	config, ok := conf.(model.ScaleHost)
+	if !ok {
+		return http.StatusInternalServerError, fmt.Errorf("Can't process config")
+	}
+
+	if config.Action == "" {
+		return http.StatusBadRequest, fmt.Errorf("Scale action not provided")
+	}
+
+	if config.Action != "up" && config.Action != "down" {
+		return http.StatusBadRequest, fmt.Errorf("Invalid action %v", config.Action)
+	}
+
+	if config.Amount <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Invalid amount: %v", config.Amount)
+	}
+
+	if config.SelectorLabel == nil {
+		return http.StatusBadRequest, fmt.Errorf("Selector label not provided")
+	}
+
+	if config.Min <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Minimum scale not provided/invalid")
+	}
+
+	if config.Max <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Maximum scale not provided/invalid")
+	}
+
+	if config.Min >= config.Max {
+		return http.StatusBadRequest, fmt.Errorf("Max must be greater than min")
+	}
+
+	if config.DeleteOption != "mostRecent" && config.DeleteOption != "leastRecent" {
+		return http.StatusBadRequest, fmt.Errorf("Invalid delete option %v", config.DeleteOption)
+	}
+
+	selectorLabel := make(map[string]string)
+	for key, value = range config.SelectorLabel {
+		selectorLabel[key] = value
+	}
+
+	cattleConfig := rConfig.GetConfig()
```

Comment:
Just confirming that this comment was for `ValidatePayload`. I have changed that function to use the actual client